--- a/package/boot/arm-trusted-firmware-airoha/scripts/airoha_pack_bl2.sh
+++ b/package/boot/arm-trusted-firmware-airoha/scripts/airoha_pack_bl2.sh
@@ -60,7 +60,7 @@ cat $BL22 >> $BL2
 cat $BL23 >> $BL2
 cat $FLASH_TABLE >> $BL2
 
-crc=$(cksum -a crc32b $BL2 | cut -d' ' -f1)
+crc=$(python3 -c "import zlib; print(zlib.crc32(open('$BL2', 'rb').read()))")
 
 # XOR with 0xffffffff
 crc=$((0xffffffff ^ $crc))
